Configuration reload fails two times before successful reload

Hi,

when I edit and save a config file inside the conf folder I see two IOExceptions before the reload is successful.

```java
2017-03-30 13:59:58.673 [ERROR] [.core.internal.folder.FolderObserver] - Error handling update of file '/var/packages/openHAB2/target/src/conf/sitemaps/admin.sitemap': File '/var/packages/openHAB2/target/src/con
f/sitemaps/admin.sitemap' cannot be read.
java.io.IOException: File '/var/packages/openHAB2/target/src/conf/sitemaps/admin.sitemap' cannot be read
    at org.apache.commons.io.FileUtils.openInputStream(FileUtils.java:251)[30:org.apache.commons.io:2.2.0]
    at org.eclipse.smarthome.model.core.internal.folder.FolderObserver.checkFile(FolderObserver.java:239)[123:org.eclipse.smarthome.model.core:0.9.0.201703201701]
    at org.eclipse.smarthome.model.core.internal.folder.FolderObserver.processWatchEvent(FolderObserver.java:308)[123:org.eclipse.smarthome.model.core:0.9.0.201703201701]
    at org.eclipse.smarthome.core.service.WatchQueueReader.run(WatchQueueReader.java:209)[99:org.eclipse.smarthome.core:0.9.0.201703201701]
    at java.lang.Thread.run(Thread.java:745)[:1.8.0_121]
2017-03-30 13:59:58.675 [ERROR] [.core.internal.folder.FolderObserver] - Error handling update of file '/var/packages/openHAB2/target/src/conf/sitemaps/admin.sitemap': File '/var/packages/openHAB2/target/src/con
f/sitemaps/admin.sitemap' cannot be read.
java.io.IOException: File '/var/packages/openHAB2/target/src/conf/sitemaps/admin.sitemap' cannot be read
    at org.apache.commons.io.FileUtils.openInputStream(FileUtils.java:251)[30:org.apache.commons.io:2.2.0]
    at org.eclipse.smarthome.model.core.internal.folder.FolderObserver.checkFile(FolderObserver.java:239)[123:org.eclipse.smarthome.model.core:0.9.0.201703201701]
    at org.eclipse.smarthome.model.core.internal.folder.FolderObserver.processWatchEvent(FolderObserver.java:308)[123:org.eclipse.smarthome.model.core:0.9.0.201703201701]
    at org.eclipse.smarthome.core.service.WatchQueueReader.run(WatchQueueReader.java:209)[99:org.eclipse.smarthome.core:0.9.0.201703201701]
    at java.lang.Thread.run(Thread.java:745)[:1.8.0_121]
2017-03-30 13:59:58.971 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'admin.sitemap'
2017-03-30 13:59:58.988 [INFO ] [el.core.internal.ModelRepositoryImpl] - Refreshing model 'admin.sitemap'
```
I am editing the files directly from command line using vi.

Regards
Ingo

--- a/bundles/model/org.eclipse.smarthome.model.core/src/main/java/org/eclipse/smarthome/model/core/internal/folder/FolderObserver.java
+++ b/bundles/model/org.eclipse.smarthome.model.core/src/main/java/org/eclipse/smarthome/model/core/internal/folder/FolderObserver.java
@@ -238,6 +238,9 @@ public class FolderObserver extends AbstractWatchService implements ManagedServi
                         if (parsers.contains(getExtension(file.getName()))) {
                             try (FileInputStream inputStream = FileUtils.openInputStream(file)) {
                                 modelRepo.addOrRefreshModel(file.getName(), inputStream);
+                            } catch (Exception e) {
+                                LoggerFactory.getLogger(FolderObserver.class)
+                                        .warn("Error while opening file during update: {}", file.getAbsolutePath());
                             }
                         } else {
                             ignoredFiles.add(file);

--- a/bundles/model/org.eclipse.smarthome.model.core/src/main/java/org/eclipse/smarthome/model/core/internal/folder/FolderObserver.java
+++ b/bundles/model/org.eclipse.smarthome.model.core/src/main/java/org/eclipse/smarthome/model/core/internal/folder/FolderObserver.java
@@ -12,6 +12,7 @@ import static java.nio.file.StandardWatchEventKinds.*;
 import java.io.File;
 import java.io.FileInputStream;
 import java.io.FilenameFilter;
+import java.io.IOException;
 import java.nio.file.Path;
 import java.nio.file.WatchEvent;
 import java.nio.file.WatchEvent.Kind;
@@ -238,7 +239,7 @@ public class FolderObserver extends AbstractWatchService implements ManagedServi
                         if (parsers.contains(getExtension(file.getName()))) {
                             try (FileInputStream inputStream = FileUtils.openInputStream(file)) {
                                 modelRepo.addOrRefreshModel(file.getName(), inputStream);
-                            } catch (Exception e) {
+                            } catch (IOException e) {
                                 LoggerFactory.getLogger(FolderObserver.class)
                                         .warn("Error while opening file during update: {}", file.getAbsolutePath());
                             }

